src/base/Identifier.ml

```diff
@@ -19,21 +19,89 @@
 open Core_kernel
 open ErrorUtils
 
-type 'rep t = Ident of string * 'rep [@@deriving sexp]
+module type QualifiedName = sig
 
-let asId i = Ident (i, dummy_loc)
+  type t [@@deriving sexp]
 
-let asIdL i loc = Ident (i, loc)
+  val as_string : t -> string
 
-let get_id i = match i with Ident (x, _) -> x
+  val as_error_string : t -> string
 
-let get_rep i = match i with Ident (_, l) -> l
+  val equal_id : t -> t -> bool
+  
+  val compare_id : t -> t -> int
```

Comment:
Perhaps we should reuse the standard names here? Namely, `( = )` and `compare`.

Reply:
Then we could just say 
```
type t [@@deriving sexp, compare, equal]
```
and remove `equal_id` and `compare_id`.

Reply:
Good idea. I'll fix it.

Reply:
`[@@deriving equal]` generates the function `equal` rather than the function `=`, so I've gone with that.

Reply:
Right, sorry for the confusion.